### ISSUE TITLE
Native compile on arm gives me "libavutil/avutil.h": File does not exist"

### ISSUE THREAD

**Issue by mpandrews2000** [bytedeco/javacpp-presets#322]
Hi,

I'm trying to do a native ARM build on a Jetson TK1, bt I can't get around the error:

```
[INFO] Targeting /home/ubuntu/builds/javacpp-presets/ffmpeg/src/main/java/org/bytedeco/javacpp/avcodec.java
[ERROR] Failed to execute JavaCPP Builder: Could not parse "libavutil/avutil.h": File does not exist
```

even though libavutil/avutil.h is clearly under the ffmpeg directory.

Any help would be greatly appreciated.

**Comment by saudet** [comment]
Make sure to specify manually the platform as per the guide here:
https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments#linux-arm

**Comment by mpandrews2000** [comment]
Hi

I am not tying to cross compile, I am trying to compile directly on the TK1.

- Paul

On Nov 23, 2016, at 4:22 AM, Samuel Audet <notifications@github.com<mailto:notifications@github.com>> wrote:

Make sure to specify manually the platform as per the guide here:
https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments#linux-arm

-

**Comment by saudet** [comment]
Yes, please try anyway. Platform detection has not been tested for Linux on
ARM.

**Comment by saudet** [comment]
@vb216 Any comments about this?

**Comment by vb216** [comment]
Cool board, not seen that one before. Will take a look.. I did have in mind, that it could be a good idea to rework the armhf build so that it would compile natively on arm and have some aliases/symlinks in the cross compile docker so that the same build carries on working there too

**Comment by vb216** [comment]
Getting there.. so far only real changes have been to ffmpeg cppbuild.sh. OpenBLAS took a long time to build, and spotted something up with the flags - multiple definitions of march and mfpu, looks like it picks up both definitions of platform.compiler.fastfpu and platform.compiler.default ? It's not a problem as such as still builds, just seems odd and less certainty over which are definitely going to be used

[INFO] g++ -I/home/pi/javacpp-presets/openblas/cppbuild/linux-armhf/include/ -I/usr/lib/jvm/jdk-8-oracle-arm32-vfp-hflt/include -I/usr/lib/jvm/jdk-8-oracle-arm32-vfp-hflt/include/linux /home/pi/javacpp-presets/openblas/target/classes/org/bytedeco/javacpp/jniopenblas.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -march=armv7-a -mfpu=neon -mfloat-abi=hard -ffast-math -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -shared -o libjniopenblas.so -L/home/pi/javacpp-presets/openblas/cppbuild/linux-armhf/lib/ -Wl,-rpath,/home/pi/javacpp-presets/openblas/cppbuild/linux-armhf/lib/ -lopenblas

**Comment by vb216** [comment]
Should build natively on Pi OK now, I've tested it on a Pi3 here:
mvn install -Djavacpp.platform=linux-armhf -Dmaven.javadoc.skip=true

You'll need to add some swap for openblas most likely, and add in the usual dependencies. I need to update the wiki build instructions, as you can now either grab the public docker image and cross compile straight away, or add required build tools and libs and build native now.

**Comment by saudet** [comment]
Fix included in version 1.3, enjoy! 

Thanks for the fix @vb216 :)
